### PR TITLE
Updates to userguide

### LINKED ISSUE #90: [PE-D][Tester A] Date is not showing if one part is missing
**Command executed**

**Result**

**Command executed**

**Result**

Sometimes, what the users care about the most is the application deadline. Therefore, it might be the case where only the deadline is recorded.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: JustWeiHao/ped#4

### LINKED ISSUE #93: [PE-D][Tester D] Duplicates when adding companies with just a start date or an end date
How to reproduce?
1. `add -n DBS -t Software Engineer -e dbs@example.com`
2. `add -n DBS -t Software Engineer -e dbs@example.com -d1 2024-04-04`
3. `add -n DBS -t Software Engineer -e dbs@example.com -d2 2024-05-05`
4. All of these commands will give rise to 3 identical inputs on the list, with no date reflected.

Potential problems:
Users may not be able to differentiate that it is actually 3 separate listings. Perhaps, you could consider having a start date only if there is a deadline field? Like allow '-d1 -d2', '-d2' and no date, but not '-d1' alone, and can check for duplicates.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: bertrandong/ped#2

### LINKED ISSUE #96: [PE-D][Tester A] Two companies with same email
Considering the real-world context, I believe it is unlikely to happen right?

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: JustWeiHao/ped#6

### LINKED ISSUE #97: [PE-D][Tester D] Add function currently allows dates that have already passed as inputs
How to reproduce?
1. `add -n DBS -t Software Engineer -e dbs@example.com -d1 2023-04-04 -d2 2023-05-05`

Potential Problems:
- Users may be looking at data that is no longer relevant for them

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: bertrandong/ped#3

### LINKED ISSUE #101: [PE-D][Tester A] Suspicious equality between two job postings
**Same position from the same company but different time range**

**Almost the same, just that one has a phone number**

It is unlikely that a company will open the same role with overlapping but different time ranges.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: JustWeiHao/ped#8

### LINKED ISSUE #115: [PE-D][Tester B] start date can be inputted without end date, and vice versa
We are allowed to key in start and end dates without the other half, which results in a valid entry that does not display the date.
Would recommend fixing this  because the internbook recognizes this as a separate input from one with no date keyed in.

-------------
Labels: `type.FeatureFlaw` `severity.High`
original: Ragnapop/ped#11

### LINKED ISSUE #127: [PE-D][Tester E] Sorting by date has unexplained behavior when sorting companies with no date
I ran the command `sort a and then `sort e` to sort by end date. It pushed all the companies to the bottom of the list, but in undefined order (not in alphanumeric order either as would be expected). This is not necessarily a functionality bug as pushing all these companies to the bottom of the list makes sense. However, it would be better if this behavior is explained in the UG so that users can expect this behavior. It's really a small bug, just more for consistency.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: pjanthony2001/ped#14

### PR DESCRIPTION
**Pull request by SherwynNg**
Provide more details regarding the issues, addressing users' concerns and confusion

Closes #90 
Closes #93 
Closes #96 
Closes #97 
Closes #101 
Closes #115 
Closes #127

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -364,24 +368,62 @@ If your changes to the data file makes its format invalid, InternBook will disca
...
+**A**: Yes, it's possible. We permit multiple company names to share the same email address. For instance, consider 
+inputs like OCBC Tech or OCBC Finance. We offer users the flexibility to customise company names, including departmental 
+distinctions, even if they share an email address.
+
+**Q**: Why is it possible to add applications with dates that have already passed?
+**A**: Users are permitted to add applications with past dates because we're in the process of developing new 
+[features](#future-advancements). These features will enable users to take notes about specific companies, which they can
+refer back to when considering future applications.
+
+**Q**: When is a company classified as a duplicate?
```
Comment (AjayArvind2207) [inline docs/UserGuide.md:422]: Similarly, you can consider adding a `<br>` here

**Review by AjayArvind2207 — COMMENTED** [review]
LGTM other than some minor formatting nits!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -184,6 +184,10 @@ Examples:
 Remark:
 * We allow the adding of duplicate companies with different start dates and end dates.
 * This is fully intended and not a bug as this can allow users to keep track of different application windows of the same role in the same company.
+* Check out our [FAQ](#faq) to understand how we distinguish duplicate entries.
+* Start date can be added without the end date, vice versa. However, when both dates are not entered, the record will
```
Comment (AjayArvind2207) [inline docs/UserGuide.md:188]:
```suggestion
* Start date can be added without the end date, and vice versa. However, when both dates are not entered, the record will
```

**Review by AjayArvind2207 — COMMENTED** [review]
One last minor nit but otherwise LGTM!

File: docs/UserGuide.md [hunk 3]
```diff
@@ -184,6 +184,10 @@ Examples:
 Remark:
 * We allow the adding of duplicate companies with different start dates and end dates.
 * This is fully intended and not a bug as this can allow users to keep track of different application windows of the same role in the same company.
+* Check out our [FAQ](#faq) to understand how we distinguish duplicate entries.
+* Start date can be added without the end date, and vice versa. However, when both dates are not entered, the record will
```
Comment (Xilef121) [inline docs/UserGuide.md:188]: I believe in the case of our app, it is "when either dates are not entered" rather than "when both dates are not entered"

Comment (Xilef121) [reply docs/UserGuide.md:188]:
```suggestion
* Start date can be added without the end date, and vice versa. However, when either dates are not entered, the record will
```

**Review by cocoanautz — COMMENTED** [review]
LGTM

**Pull request merged**